core/src/main/scala/com/permutive/refreshable/Refreshable.scala

```diff
@@ -217,14 +252,30 @@ object Refreshable {
       } yield RefreshableImpl(store, fiberStore, makeFiber(store))
     }
 
+    private def storeValue(
+        store: Ref[F, CachedValue[A]],
+        oldValue: CachedValue[A],
+        newValue: CachedValue[A]
+    ): F[Unit] =
+      combineFunction.fold(store.set(newValue)) { f =>
+        f(oldValue, newValue).flatMap { v =>
```

Comment:
I guess we should update the `Ref` via `guaranteeCase` here? If `f` throws then we would never update

Reply:
I was thinking about this and was wondering whether we should use gurantee, but then thought that because this function returns a new value, we are dependent on that value and should blow up if `f` throws. It's a bit of a "use at your own risk" and if your function doesn't return a valid value then the cache won't be updated - do the error handling yourself.